lib/ansible/modules/windows/win_domain_computer.ps1

```diff
@@ -0,0 +1,186 @@
+#!powershell
+#
+# Copyright: (c) 2017, AMTEGA - Xunta de Galicia
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+
+# ------------------------------------------------------------------------------
+$ErrorActionPreference = "Stop"
+
+# Preparing result
+$result = @{}
+$result.changed = $false
+
+# Parameter ingestion
+$params = Parse-Args $args -supports_check_mode $true
+
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool"  -default $false
+$diff_support = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+
+$name = Get-AnsibleParam -obj $params -name "name" -failifempty $true -resultobj $result
+$sam_account_name = Get-AnsibleParam -obj $params -name "sam_account_name" -default "$name$"
+If (-not $sam_account_name.EndsWith("$")) {
+  Fail-Json -obj $result "sam_account_name must end in $"
+}
+$enabled = Get-AnsibleParam -obj $params -name "enabled" -type "bool" -default $true
+$description = Get-AnsibleParam -obj $params -name "description" -default "" -resultobj $result
+$state = Get-AnsibleParam -obj $params -name "state" -ValidateSet "present","absent" -default "present"
+If ($state -eq "present") {
+  $dns_hostname = Get-AnsibleParam -obj $params -name "dns_hostname" -failifempty $true -resultobj $result
```

Comment:
You don't have to change this but you could put the `Get-AnsibleParam` outside of the if statement and still have the `failifempty` by doing

`$dns_hostname = Get-AnsibleParam -obj $params -name "dns_hostname" -failifempty ($state -eq "present")`

Reply:
Interesting. I will keep it unchanged for now unless told otherwise.